catroid/src/main/java/org/catrobat/catroid/utils/TextSizeUtil.java

```diff
@@ -69,6 +80,38 @@ public static void enlargeViewGroup(ViewGroup viewGroup) {
 		}
 	}
 
+	public static void enlargeTableLayoutButtonText(TableLayout tableLayout) {
+		if (modifier == 1.0f) {
+			return;
+		}
+
+		Context context = CatroidApplication.getAppContext();
+		WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+		Display display = wm.getDefaultDisplay();
+		DisplayMetrics metrics = new DisplayMetrics();
+		display.getMetrics(metrics);
+
+		if ((metrics.widthPixels / metrics.xdpi) > widthForIncreasedTextSizeInInches) {
+			modifier += modifierIncreaseDelta;
```

Comment:
I increase the modifier by the delta, resize the text, and then decrease the modifier by the delta so it's back to normal again. This happens in the same function.
Unless i misunderstood you, I don't think this could pose a problem?

I changed the if statement to check a boolean for clarity, and moved the code to determine if it's a large device in a separate function.